badge_server/badge_server.py

```diff
@@ -222,7 +284,44 @@ def index():
     return 'Visitor number: {}'.format(value)
 
 
-@app.route('/self_compatibility_badge/image')
+@app.route('/one_badge_image')
+def one_badge_image():
+    package_name = flask.request.args.get('package')
+    url_prefix = flask.request.url_root[:-1]
+    # Call the url for each badge to run the checks.
+    # Self compatibility badge
+    requests.get(url_prefix + flask.url_for(
+        'self_compatibility_badge_image', package=package_name))
+    # Google compatibility badge
+    requests.get(url_prefix + flask.url_for(
+        'google_compatibility_badge_image', package=package_name))
+    # Self dependency badge
+    requests.get(url_prefix + flask.url_for(
+        'self_dependency_badge_image', package=package_name))
+
+    status, _, _, _ = _get_all_results_from_cache(package_name)
+    color = STATUS_COLOR_MAPPING[status]
+    package_name = package_name.replace('-', '.')
+    url = URL_PREFIX + '{}-{}-{}.svg'.format(package_name, status, color)
+
+    return requests.get(url).text
+
+
+@app.route('/one_badge_target')
+def one_badge_target():
+    package_name = flask.request.args.get('package')
```

Comment:
Don't you have to populate the cache here?

Reply:
This route is for generating the package detail page, which uses the cached result calculated by calling `one_badge_image`. We don't need to populate the cache here. So this must run after `one_badge_image` is called (and github badges works like this), or the package page will show an empty status.